packages/http/src/mocker/generator/__tests__/JSONSchema.spec.ts

```diff
@@ -9,15 +9,37 @@ describe('JSONSchema generator', () => {
         properties: {
           name: { type: 'string', minLength: 1 },
           email: { type: 'string', format: 'email' },
+          ip: { type: 'string', 'x-faker': 'internet.ip' },
         },
-        required: ['name', 'email'],
+        required: ['name', 'email', 'ip'],
       };
 
       const instance = generate(schema);
       expect(instance).toHaveProperty('name');
       expect(instance).toHaveProperty('email');
     });
 
+    it('generates dynamic example with faker from schema', () => {
+      const schema: JSONSchema = {
+        type: 'object',
+        properties: {
+          name: { type: 'string', minLength: 1 },
+          email: { type: 'string', format: 'email' },
+          ip: { type: 'string', 'x-faker': 'internet.ip' },
+        },
+        required: ['name', 'email', 'ip'],
+      };
+
+      const instance = generate(schema);
+      expect(instance).toHaveProperty('name');
+      expect(instance).toHaveProperty('email');
+      expect(instance).toHaveProperty('ip');
+      // @ts-ignore
+      expect(instance.ip).not.toBeUndefined();
+      // @ts-ignore
+      expect(instance.ip).not.toEqual('internet.ip');
```

Comment:
I have some questions :D 

What errors were you getting that made you add the ts.ignore? I'm certainly no TS expert but I'd like to see if we can avoid it. 

Also, this test is proving something that is never gonna happen anyway right?

I think what would be excellent here, is running this schema but just `ip: { type: 'string' }` then find some sort of "is IP address" assertion which will fail because json schema faker is going to put "cheese" "monkies" "lorum" etc into the value. 

Then you slide the `x-faker` keyword in:

```
schema.properties.ip['x-faker'] = 'internet.ip';
```

Now run it again and assert that it now _is_ an IP address.

This will prove that json schema faker is indeed leveraging faker to do its job and its spotting the keyword. A perfect integration test.

Reply:
@philsturgeon The reason for that lines is that the responses are not typed; if you want them typed, you need to validate them appropriately, thing that we're not doing now. We *could* use io-ts, but not now. I want to introduce it one day though.